week1/homework/src/server.js

```diff
@@ -8,10 +8,10 @@ const http = require('http');
 function createServer(port) {
   let state = 10;
 
-  function respond(response) {
-    response.writeHead(200, {'Content-Type':'application/json'});
-    response.write(JSON.stringify({state}));
-    response.end();
+  function respond(status) {
```

Comment:
why did you end up changing this parameter's name? It was fine :)
What I meant is to pass an additional parameter to the function to avoid reading a variable from outside the function's scope.

```js
function respond(response, stateObject) {
  response.writeHead(200, {'Content-Type':'application/json'});
  response.write(JSON.stringify(stateObject));
  response.end()
}
```

And you would call it like:
```js
respond(response, { status });
```

does it make sense?

Reply:
Oh I thought the parameters name could make problem, that"s what I thought you meant.